ql/src/java/org/apache/hadoop/hive/ql/exec/SerializationUtilities.java

```diff
@@ -228,6 +230,27 @@ public void setConf(Configuration conf) {
     public Configuration getConf() {
       return configuration;
     }
+
+    @Override
+    public com.esotericsoftware.kryo.Registration getRegistration(Class type) {
```

Comment:
Just to make sure my understanding is correct, is this method called before the class being instantiated? So, we can avoid some classes be instantiated.

Reply:
I think so, `getRegistration` is place for check whether an unregistered class is encountered when `registrationRequired` is set to true: 
https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/Kryo.java#L552-L556
However we cannot simply enable `registrationRequired` for UDF deserialization, as non-transient fields in the UDF are arbitrary, it's difficult to confine the inputs.
Thank you for the review and comments!